### ISSUE TITLE
questions about ergan project

### ISSUE THREAD

**Issue by yahavhalpern** [Bingwen-Hu/ERGAN-Pytorch#7]
hi,
thanks for sharing your code it is very helpful i have few questions about the flow:
1. what do you mean by:" If you want to test your own data with our pre-trained model, you need to align the data first ( refer to CelebA) or retrain with your own data." Does the model works only on the data it has been trained on? the alignment is to resize the image so the eyes will match the mask? i tried to take a photo of my own but just got blurry mask on top of the photo
2. can yo refer me to the section in the code where you teach the model to insert eyes instead of glasses? it works only in case the glasses are directly in the mask frame?so if i have two photos one with glasses and one with not but their face pose is not the same; lets say the one without is looking to the left and the one with glasses is turning right then the match wont be good?
Thank you
Yahav
![WhatsApp Image 2021-09-14 at 12 30 19](https://user-images.githubusercontent.com/82221517/133303135-74966237-33e2-470a-bc2d-78a24c13208e.jpeg)
![WhatsApp Image 2021-09-14 at 12 30 31](https://user-images.githubusercontent.com/82221517/133303136-5b6dfe00-e9ea-46ea-9daa-9f659c6d887c.jpeg)
![WhatsApp Image 2021-09-14 at 12 31 08](https://user-images.githubusercontent.com/82221517/133303141-044fe8ac-21bc-46ca-ad1a-107efe72661e.jpeg)

**Comment by Bingwen-Hu** [comment]
Hi, Yahav,
Q1: The selection of the eye region can be flexibly intercepted according to the aligned face image. For the provided pre-training model, the eye region is determined by the alignment of CelebA dataset.  Please check out https://github.com/Bingwen-Hu/ERGAN-Pytorch/blob/0abf184ef5b744e9d8b192fc85fb050c16e43a9e/networks.py#L163
Q2: When the input pairs of images have the same facial pose (whether positive or not),  our method outputs the optimal result, and in other cases, it outputs the suboptimal result.
